### DISCUSSION TITLE
Out of stock parts

### DISCUSSION THREAD

**Discussion by rszemeti** [peterantypas/maiana#49] [General]
Is there a replacement for the AP63203WU-7 ... I can't find these anywhere ...  Octopart shows everyone is out of stock.

the 8 way TVS diode is another "problem" part ...

**Comment by peterantypas** [comment]
I'm afraid not, and I tried to look for one ... if this doesn't get
resolved, the design is not viable :(

**Comment by rszemeti** [comment]
It looks like they expect stock around 3rd March for some suppliers ... it
is a popular part, I think it will be OK .. anyway, I have 5 boards and all
the other parts ... so it has to be OK ;)

**Comment by peterantypas** [comment]
I've pretty much given up on this quarter. I expect things to start
returning to normal by the end of Q2, so not planning any production on
this or my other projects until then.

**Comment by peterantypas** [comment]
I am redesigning the board with the AP3211 buck (the original) and the LDFPVR (2x2mm) LDO. Both are widely available. 

I believe I have secured enough RF components to make a few kits in the next quarter. Of course, some of them are coming from shady spot markets in Shenzhen, so we'll see ...

**Comment by rszemeti** [comment]
I already have 5 x V10 boards here, so I'll just wait for the bits to come
into stock ... looks like should be early March or so.

I was thinking of running all 5, and making one into an AIS for the dinghy,
useful for when we lose it ...  I might put one in our liferaft ...

**Comment by rszemeti** [comment]
Farnell now have stock of the regualtor:

https://uk.farnell.com/search?st=ap63203wu-7&sp=y

Just need to find time now and set up to assemble these 5.  I've been bored in lockdown and playing with my SMD machine, I've figured out a way to build small projects, without buying reels of parts.  Basically, I've written some Perl to parse the Eagle .BRD file and write a production program for the machine ... the common parts it pulls off reels as normal, but the "specials", it prints out a PDF with the cut-strip components laid out on it.  I stick it down on a piece of board, and then stick the component strips on top .. put it in the machine and peel back the plastic covers .... and the program is set up to treat them as "tray feeders", that just happen to be 8mm wide and with a 2mm or 4mm pitch.  Should be perfect for those one offs that are just too boring to assemble by hand!

**Reply by peterantypas** [reply]
I would love to see a picture of this in action

**Reply by revolteur** [reply]
Cool and Same here. Would love to see how this is done as well.  Which SMD machine are you using?

**Comment by rszemeti** [comment]
Zevatech 570L  .. I think it was sold as the Juki KP620E in the USA

I have a video somewhere of when I first got the software running, but not
with the sheet of paper component matrix  ... I won't bore you with the
details, but basically, I don't have the factory offline programming
software, so I wrote my own and spent tens and tens of hours reverse
engineering the undocumented binary file format ...  I'll take a decent
video of the process from pasting up the board to running it through the
oven when I do these boards.

**Comment by rszemeti** [comment]
Is the STM32L422KBU6 a suitable alternative?  Looks like it should be OK.  really struggling on the 412 and the 432 variants at the moment ... and now the SMD antenna appears to have gone out of stock everywhere as well .. doh :)

**Comment by peterantypas** [comment]
I don't see why it wouldn't work, but the build will need some touch-up.

**Reply by peterantypas** [reply]
That said, DigiKey has plenty of L412s right now: https://www.digikey.com/en/products/detail/STM32L412K8U6/497-STM32L412K8U6-ND/10414668?itemSeq=355557148

**Comment by rszemeti** [comment]
The shipping from USA is much more than the parts, digikey is a really
difficult supplier here in the UK, it takes them a long time to get here.
I will think about it a bit more ... maybe the $20 of shipping will be less
than the time adjusting be build ...

**Comment by peterantypas** [comment]
Let me see if I can get it to build for that target with no errors at least
... give me a few

**Comment by peterantypas** [comment]
Check out branch L422, see if it works for you, then we'll merge it to
master. It builds without errors and the clock configuration looks sane.

The linker script is deliberately shared with the L412 to restrict RAM to 39K and flash to 62K.

**Comment by rszemeti** [comment]
Jolly good, I shall order the last bits then, I was just going through the
full BOM and working out which inductors I had missed ... last Farnell
order going in in the morning, then I'll paste up the boards and give it a
go!

I've been busy working on NMEA2000 stuff (as I am still not allowed to
visit my boat, I have to do something, right?)  Using Timo Lappalainen's
excellent library,  I have a dual-channel battery monitor, an engine
monitor,  compass/gyro roll sensor, a Pi "hat" with NMEA2K and serial ..
and an interface to produce  3 phase "wind direction" signal and "pulses"
for wind speed/boat speed so I can drive my old instruments from the new
sensors and NMEA2000 bus ... working hard, I hope to send the gerbers for
the boards at the end of the week ... plus of course some boards for work.
I just need to decide on exactly which box I put them in.    So far just
prototyped with some Aduino boards connected to my BYG sensors sitting in
my living room, but, all seems to work.

Maybe I should do a board to take the RJ45 breakout into NMEA2k?  It would
be an easy way to integrate Mianana into the boat systems..... most plotter
software just needs the data and will do the complex part of deciding about
alarms itself.

[image: image.png]

**Reply by peterantypas** [reply]
RJ45 -> NMEA2k sounds like a great idea to me

**Comment by rszemeti** [comment]
I'll try and make a simple board ... I included a USB connector because it
makes it "easy" for Arduino programming,  but it could be removed ..

Do you think it is worth keeping the USB connector for serial, so people
can set MMSI number, boat name etc from a laptop?  I'll swap that huge CAN
chip for a QFN package too!

**Reply by peterantypas** [reply]
I think so. It's always handy to have for troubleshooting.

**Reply by peterantypas** [reply]
On that note, I want to think about building different breakouts PCBs as simple adapters to different interfaces, without fancy enclosures and LEDs, etc. I guess they could have status LEDs too, but let the user decide where to enclose them and wire them. Each boat is different ...

**Comment by peterantypas** [comment]
Spot market did deliver, but at a cost near 2x:

![IMG_1034](https://user-images.githubusercontent.com/1565933/111342392-a84eec80-8637-11eb-9eef-9d4454485e5f.jpg)

At least I can start thinking about producing the first 10 kits.

**Reply by Peter-Petrik** [reply]
Excellent project, and very much interested in obtaining one of these kits as they become available. Let me know if there's a specific sign-up list and/or payment information.

**Reply by peterantypas** [reply]
The AP6302xxx series is OOS everywhere. The AP3211 is available in spot markets so I was able to order enough. 

The kit will feature board rev 11 which is being redesigned. The main changes are:

- AP3211 as a buck switch for 3.3V
- Additional filtering on the 3.3V rail
- LDFPVR instead of LDFPUR for the PA voltage regulator 
- LNA power comes from a freed MCU GPIO (deleted DMG1016UDW)
- GreenPAK IC to consolidate several I/O functions (deleted the 3 DMN63D8LDW)
- Matching network for the GNSS antenna (see below) 

After designing a board with a proper RF switch on the GNSS trace, I was able to determine that it was off in terms of tuning, so it need a 12pF series cap to hit the center (at least for GPS):

![Board10-GPS-S11-12pfSeries](https://user-images.githubusercontent.com/1565933/111659855-c7ca4e80-87ca-11eb-88c9-465525b95d0f.png)

I think this is OK for GLONASS too, one cannot expect really good tuning for both bands.

**Comment by rszemeti** [comment]
That's come in quite nicely.  Very useful things those nano VNA, I've been
using the NWT analyser/tracking generator scalar analyser thing,  similar
sort of idea but goes up to 4.4GHz.  Been too busy on other projects to
make much progress on this at the moment, did a nice 300W LDMOS PA for the
70MHz band, modeled in SimSmith and the build came out exactly as
predicted, which was nice!

If you haven't tried SimSmith for Smith chart stuff, have a look, it is
excellent.  Of course, a copy of MicrowaveOffice would be nicer ... ;)

**Comment by Rudder2** [comment]
Awesome! We are moving forward! Can't wait for the kit to become available either. I'm planning on using this as my ships main GPS and AIS for my OpenPlotter setup.

**Comment by peterantypas** [comment]
Test jig and golden board are done, so now waiting on the BOM to start building ...

![IMG_1046](https://user-images.githubusercontent.com/1565933/113079744-7dfa4480-918a-11eb-82b1-dc194e7965f5.jpg)

**Comment by rszemeti** [comment]
Neat.  I have *almost* all of the BOM .. the poxy smps chip seemed to go
out of stock some time between me putting stuff in the Mouser cart and the
order going through ... apart from that, I think I have all the bits for 5,
not going to paste the boards and run them until I have the last bit though.

**Comment by peterantypas** [comment]
Starting SMT:

![IMG_1052](https://user-images.githubusercontent.com/1565933/113527480-8380d180-9572-11eb-8327-047db83a7d12.JPG)

**Comment by peterantypas** [comment]
I'm afraid I have some bad news. My main project (the one that pays my salary) just got dealt a major blow due to Qualcomm's supply crunch and it will require 100% of my time over the next few months to be revived. MAIANA will have to go on the backburner :(

**Reply by muebau** [reply]
Its very sad to hear that. It is very impressive what you build in here and I cant wait to see what comes next :smiley: .
All the best for your main project and all the luck needed. :wink:

**Comment by rszemeti** [comment]
Thank you for your efforts so far, the supply crunch is bad everywhere.  I
have several products that have had to have changes as part we have been
able to buy for years are now unobtainable. I have a few new products I
have struggled to even prototype as the designed-in part (mostly TI stuff)
has become unobtainable even before the boards arrived.

I'm pressing on with some other marine/NMEA2000 projects in the meantime.
It's not easy out there!
